core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/DiagnosticLog.java

```diff
@@ -346,20 +306,12 @@ public DiagnosticLog build() {
                 // required. There shouldn't be a diagnostic log without a result message and input.
                 throw new IllegalStateException("Either resultMessage or input must be provided.");
             }
-            if (this.logId == null) {
-                logId = UUID.randomUUID().toString();
-            }
-            if (this.recordedAt == null) {
-                recordedAt = Instant.now();
-            }
-            if (this.requestId == null) {
-                requestId = MDC.get(CORRELATION_ID_MDC);
-            }
-            if (this.flowId == null) {
-                flowId = MDC.get(FLOW_ID_MDC);
-            }
-            if (this.logLevel == null) {
-                logLevel = LogLevel.BASIC;
+            logId = UUID.randomUUID().toString();
```

Comment:
With this change we are removing the capability to set a log ID. Is this expected?

Reply:
we always set Log id is using `UUID.randomUUID().toString();`. Also, we are using the values of MDC to other attributes (requested, requested, flowId)